demo/src/__tests__/SvgDisplay.spec.tsx

```diff
@@ -4,52 +4,56 @@ import SvgDisplay from "../SvgDisplay";
 import mem from "../../../src/index";
 const { draw } = mem;
 
+const mockMemoryModels = {
+    serializeSVG: jest.fn(),
+    clear: jest.fn(),
+    render: jest.fn(),
+};
+
 jest.mock("../../../src/index", () => ({
-    draw: jest.fn(() => ({
-        clear: jest.fn(),
-        render: jest.fn(),
-    })),
+    draw: jest.fn(() => mockMemoryModels),
 }));
 
 describe("SvgDisplay", () => {
+    const setSvgResultMock = jest.fn();
+
     describe("when jsonResult is not null", () => {
         let jsonResult: Object;
+
         beforeEach(() => {
-            jsonResult = [
-                {
-                    isClass: true,
-                    name: "__main__",
-                    id: null,
-                    value: { lst1: 82, lst2: 84, p: 99, d: 10, t: 11 },
-                    stack_frame: true,
-                },
-                {
-                    isClass: false,
-                    name: "str",
-                    id: 19,
-                    value: "David is cool!",
-                    style: ["highlight"],
-                },
-                { isClass: false, name: "int", id: 13, value: 7 },
-            ];
-            render(<SvgDisplay jsonResult={jsonResult} />);
+            // In order to get to draw function, input has to be valid json otherwise JSON.parse fails
```

Comment:
It isn't clear why you made this change. Is the original `jsonResult` here not valid json?

Reply:
The idea was to show that input just has to be valid JSON not valid Memory Models JSON to get to the draw function - if you feel like the prev JSON was more appropriate I can change it back

Reply:
Added another `describe` suite via `describe.each` reusing the same tests. Results look like 
![image](https://github.com/david-yz-liu/memory-models-rough/assets/67441706/aad6b4d5-b104-47ae-a925-3b050f8ea30f)